src/App.tsx

```diff
@@ -1,59 +1,52 @@
-import React, { useEffect, useState } from 'react';
+import React, {
+  useEffect,
+} from 'react';
 import 'bulma/bulma.sass';
 import '@fortawesome/fontawesome-free/css/all.css';
 import './App.scss';
-
 import classNames from 'classnames';
 import { PostsList } from './components/PostsList';
 import { PostDetails } from './components/PostDetails';
 import { UserSelector } from './components/UserSelector';
 import { Loader } from './components/Loader';
-import { getUserPosts } from './api/posts';
-import { User } from './types/User';
-import { Post } from './types/Post';
-import { Counter } from './features/counter/Counter';
+import {
+  useAppDispatch,
+  useAppSelector,
+} from './app/hooks';
+import { fetchUsers } from './features/users';
+import { fetchPosts, actions as postsActions } from './features/posts';
 
 export const App: React.FC = () => {
-  const [posts, setPosts] = useState<Post[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
-
-  const [author, setAuthor] = useState<User | null>(null);
-  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
-
-  function loadUserPosts(userId: number) {
-    setLoaded(false);
+  const { author } = useAppSelector(state => state.users);
+  const {
+    posts,
+    isLoaded,
+    hasError,
+    selectedPost,
+  } = useAppSelector(state => state.posts);
+  const dispatch = useAppDispatch();
 
-    getUserPosts(userId)
-      .then(setPosts)
-      .catch(() => setError(true))
-      // We disable the spinner in any case
-      .finally(() => setLoaded(true));
-  }
+  useEffect(() => {
+    dispatch(fetchUsers());
+  },
+  []);
 
   useEffect(() => {
-    // we clear the post when an author is changed
-    // not to confuse the user
-    setSelectedPost(null);
+    dispatch(postsActions.setSelectedPost(null));
 
     if (author) {
-      loadUserPosts(author.id);
-    } else {
-      setPosts([]);
+      dispatch(fetchPosts(author.id));
     }
-  }, [author?.id]);
+  }, [author]);
```

Comment:
<img width="834" alt="Screenshot 2023-12-11 at 14 06 13" src="https://github.com/mate-academy/react_redux-list-of-posts/assets/135232001/74b4b1ae-587d-4b0e-96f0-ec29aabd93d7">